scipy/special/tests/test_hypergeometric.py

```diff
@@ -23,6 +23,41 @@ def test_special_cases(self):
     def test_nan_inputs(self, a, b, x):
         assert np.isnan(sc.hyperu(a, b, x)) == np.any(np.isnan([a, b, x]))
 
+    @pytest.mark.parametrize(
+        'a,b,x,expected',
+        [(1e-15, 1.0, 1e-15, 1.0000000000000346),
```

Comment:
I think negative `a` should be included in the tests. We might find that there is still a rough edge as `a` approaches `-1` because it will call `hyperu` with `a + 1`. If this can recurse the call  to `hyperu`, I think it would work. 
(If that won't work, we might need to adjusting the criterion `abs(a) < 1` to something like `abs(a) < 0.9`, but it might leave a sliver of the problematic behavior unless we fine-tune the threshold), or we could use the recurrence relation in the other direction, which I'd need to test.)